addons/mail/models/mail_thread.py

```diff
@@ -843,7 +837,17 @@ def _routing_check_route(self, message, message_dict, route, raise_exception=Tru
                 check_result = self.env['mail.alias.mixin']._alias_check_contact_on_record(obj, message, message_dict, alias)
             if check_result is not True:
                 self._routing_warn(_('alias %s: %s') % (alias.alias_name, check_result.get('error_message', _('unknown error'))), message_id, route, False)
-                self._routing_create_bounce_email(email_from, check_result.get('error_template', _generic_bounce_body_html), message)
+                _generic_bounce_body_html = _("""Hello,
+</br>The following email sent to %s cannot be accepted because this is a private email address.
+Only %s can contact us at this address.""") % (message_dict.get('to'), alias.alias_contact)
+                template = self.env.ref('mail.mail_bounce_alias', raise_if_not_found=True)
+                print(html2plaintext(alias.alias_bounced_content).strip())
+                body = template.render({
+                    'body': bool(html2plaintext(alias.alias_bounced_content).strip()) and alias.alias_bounced_content or _generic_bounce_body_html,
```

Comment:
bool(html2plaintext(alias.alias_bounced_content).strip() -> is it to try to catch void or html containing only stuff like "<p><br/></p" "? If that's the case somehow I think we could make a tool method in tools/mail.py (is_html_empty). Having a way to know html is empty even when web editor adds invisible stuff is probably a good idea.

Reply:
Yes it is the purpose. I didn't test all cases, this trick working fine. But agree, I will put a cleaner method in the tools/mail.py